fs/entry.go

```diff
@@ -103,24 +103,18 @@ func (e Entries) Update(newEntry Entry) Entries {
 		return e[i].Name() >= name
 	})
 
-	// append at the end
-	if pos >= len(e) {
-		return append(append(Entries(nil), e...), newEntry)
+	insert := 1 // insert
```

Comment:
The use of the value of `insert` for indexing below is a little subtle, but apart from that it is straight forward.